pkg/drift/gcs/gcs.go

```diff
@@ -64,24 +64,28 @@ func (c *Client) FilesWithName(ctx context.Context, bucket, filename string) ([]
 	return uris, nil
 }
 
-// DownloadFileIntoMemory fetches the file from GCS and reads it into memory.
+// DownloadAndUnmarshal fetches the file from GCS and decodes it using the
+// provided unmarshaller.
+//
 // TODO(dcreey): Handle race conditions https://github.com/abcxyz/guardian/issues/96.
-func (c *Client) DownloadFileIntoMemory(ctx context.Context, uri string) ([]byte, error) {
+func (c *Client) DownloadAndUnmarshal(ctx context.Context, uri string, unmarshaller func(r io.Reader) error) error {
```

Comment:
I think I prefer allowing any type to be passed here over passing a function to unmarshal. Then the api is pretty similar to the unmarshal api.

Reply:
I didn't want to assume that the file contents would be JSON since this is a generic gcs client. This design supports XML, raw bytes, etc.

Reply:
NIT: Gotcha. I think its probably overkill since we are doing terraform drift detection tho? Might be yagni